jdr/WFLY-13539-galleon_layer_jdr_support.adoc

```diff
@@ -0,0 +1,89 @@
+= Galleon layer for JBoss Diagnostic Reporting support
+:author:            Yeray Borges
+:email:             yborgess@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+The JBoss Diagnostic Reporting (JDR) subsystem enables the gathering of diagnostic data to aid in troubleshooting. JBoss EAP subscribers can provide this information to Red Hat when requesting support.
+
+The diagnosis information is generated by using the following operation:
+
+[source,xml]
+----
+[standalone@localhost:9990 /] /subsystem=jdr:generate-jdr-report()
+----
+
+Users are also able to generate the diagnostic information via command line executing the `jdr.sh` script located at `$WFLY_HOME/bin/` directory. This script accepts the following options:
+
+[source,bash]
+----
+$WFLY_HOME/bin/jdr.sh --help
+usage: jdr.{sh,bat,ps1} [options]
+
+JBoss Diagnostic Reporter (JDR) is a subsystem built to collect information to aid in troubleshooting. The jdr script is a utility for generating JDR reports.
+-c,--config <arg>     Configuration file of the server if it is not running.
+-h,--help             Display this message and exit
+-H,--host <arg>       hostname that the management api is bound to. (default: localhost)
+-p,--port <arg>       port that the management api is bound to. (default: 9990)
+-s,--protocol <arg>   Protocol that is used to connect. Can be remote, http or https (default: http)
+----
+
+The availability of a management interface is optional. If there is not a management interface available, the `jdr.sh` script is still able to extract some information invoking internally the embedded server.
+
+The goal of this feature is to supply a https://docs.wildfly.org/galleon/#_layers[Galleon layer] to give support to the JDR subsystem.
+
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-13539[WFLY-13539 Add a Galleon layer for JBoss Diagnostic Reporting]
+* https://issues.redhat.com/browse/EAP7-1444[EAP7-1444 Galleon Layers in EAP XP 2]
+
+=== Related Issues
+* https://issues.redhat.com/browse/WFGP-177[WFGP-177 Create a Galleon package for the JBoss Diagnostic Reporting bin scripts]
+
+=== Dev Contacts
+
+* mailto:brian.stansberry@redhat.com[Brian Stansberry]
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:fburzigo@redhat.com[Fabio Burzigotti]
+* mailto:jkasik@redhat.com[Jan Kasik]
+
+=== Testing By
+
+* [X] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+WildFly, WildFly Galleon Plugins
+
+=== Other Interested Projects
+
+== Requirements
+
+* The galleon layer must provide the provisioning of the JDR subsystem.
+* The galleon layer will depend optionally on `secure-management` layer. This layer will allow the users to configure a secure management interface to connect to the server via `jdr.sh`. If a user wants to connect via CLI, they have to add `core-tools` layer to their provision configuration.
```

Comment:
Yes, you got it right. I asked for such an example because - although IIUC a base layer is always needed :) - the snippet would definitely improve the quality of the analysis document by demonstrating what the user should do in order to use the new layer concretely and reduce the uncertainty about the steps to be followed at the same time.
Examples are definitely a great thing, IMHO :)

Reply:
I've just added an example of wildfly-jar-maven-plugin configuration using a feature pack location. The configuration will be pretty similar for all the analysis doc, but yes, it will be nice to have something as a reference for a user looking at the proposal.

Reply:
This is really nice. Just a question: since we were discussing the fact that `core-tools` is basically always there and since you mention the fact that we'll be repeating this snippet in all of the new layers proposals, WDYT about using this real example layers?
https://github.com/wildfly-extras/wildfly-jar-maven-plugin/blob/master/examples/jaxrs/pom.xml#L33

Reply:
{quote}
since we were discussing the fact that core-tools is basically always there
{quote}

`core-tools` is provisioned for all the base layers, however, on the example you pointed out provisioning `jaxrs+management`,  `core-tools` won't be available by default because none of those layers inherit `core-tools` transitively or include it by a direct layer dependency.

Reply:
Ok, thanks. Let's leave the snippet as is then, since it accounts for yet another example for the end users.